lib/actions-oclif/env/add.ts

```diff
@@ -171,31 +190,49 @@ async function setServiceVars(
 	options: FlagsDef,
 ) {
 	if (options.application) {
-		const serviceId = await getServiceIdForApp(
-			sdk,
-			options.application,
-			options.service!,
-		);
-		await sdk.models.service.var.set(serviceId, params.name, params.value!);
-	} else {
+		for (const app of options.application.split(',')) {
+			for (const service of options.service!.split(',')) {
+				try {
+					const serviceId = await getServiceIdForApp(sdk, app, service);
+					await sdk.models.service.var.set(
+						serviceId,
+						params.name,
+						params.value!,
+					);
+				} catch (err) {
+					console.error(`${err.message}, application: ${app}`);
+					process.exitCode = 1;
+				}
+			}
+		}
+	} else if (options.device) {
 		const { getDeviceAndAppFromUUID } = await import('../../utils/cloud');
-		const [device, app] = await getDeviceAndAppFromUUID(
-			sdk,
-			options.device!,
-			['id'],
-			['app_name'],
-		);
-		const serviceId = await getServiceIdForApp(
-			sdk,
-			app.app_name,
-			options.service!,
-		);
-		await sdk.models.device.serviceVar.set(
-			device.id,
-			serviceId,
-			params.name,
-			params.value!,
-		);
+		for (const uuid of options.device.split(',')) {
+			try {
+				const [device, app] = await getDeviceAndAppFromUUID(
+					sdk,
+					uuid,
+					['id'],
+					['app_name'],
+				);
+				for (const service of options.service!.split(',')) {
+					const serviceId = await getServiceIdForApp(
```

Comment:
I think it is missing a `try`/`catch`/`process.exitCode = 1` in the inner loop. The outer try/catch would catch any errors, but:
* The service name list comes from the command line, and the user could have mistyped one of the services, e.g. `--service TypoService,GoodService`. As coded, `GoodService` would not be processed as the inner loop breaks when processing `TypoService` (which would throw an error at this line).
* The overall approach in this PR is that `console.error` is printed for items that went wrong, but nothing is printed for items that succeed. Therefore, the user might incorrectly assume that `GoodService` succeeded when in fact it was not even attempted.

If a separate try/catch block is added for the inner loop, then the outer try/catch block could be reduced in scope to include only the call for `getDeviceAndAppFromUUID` (and not the inner loop). This should help a bit with readability.

Reply:
I had to leave the inner loop within the `try` block. Leaving `[device, app]` as `const` makes them unavailable in the inner loop and the linter doesn't like them being declared using `var` or `let` because their value never changes.

Reply:
I'm not sure what the code this refers to looks like but the pattern to expose something inside a try to an outer scope @nwneisen would be like
```ts
let x
try {
	x = ...;
} catch {
	x = ...; // or `throw ...` or deal with the fact x becomes `T | undefined` due to this branch
}

Reply:
@Page- Not sure how I missed trying this. I had to add an additional check for undefined variables but otherwise it is working.